### ISSUE TITLE
<p> with a style= causes error

### ISSUE THREAD

**Issue by 2020Mike** [IDPF/epubcheck#675]
For example, this paragraph:
` <p class="flush">"Time flies when you're having a good time"</p>`
Generates this error:
`Position 1`
`An error occurred while parsing the CSS: Premature end of grammar (expecting: :).`

Side note: same epub runs through Amazon Kindlegen without problems.

**Comment by mattgarrish** [comment]
You can get that error if you accidentally use style instead of class on an element:

```html
<p style="flush">...</p>
```

But I can't reproduce the error from the markup you've pasted.

**Comment by 2020Mike** [comment]
Is style not allowed in the standard?

**Comment by mattgarrish** [comment]
Sure, but it's for inlining css expressions not for adding class names. The error occurs because there's no colon after "flush" and css properties take the form "property: value", like this:

```html
<p class="flush" style="text-indent: 0">
```

The two attributes aren't interchangeable.

If kindle didn't throw an error, it's because it's not checking, but it doesn't mean your style attribute is any less broken with a class name in it.

**Comment by 2020Mike** [comment]
Inside the `<p>"Text  <span style="red">foo</span> end."</p>`

I'll change those styles to class, then I'll be done.

Strange, HTML Tidy and 'Total Validator' didn't pick pick up that I had non-CSS in a style attribute.

**Comment by mattgarrish** [comment]
Validating inlined css isn't a trivial task (or css generally, for that matter). Neither validator.nu nor validator.w3.org check either. Even epubcheck is only doing some very minimal syntax checking.
